src/Faithlife.Build/BuildRunner.cs

```diff
@@ -16,22 +16,26 @@ public static class BuildRunner
 	/// </summary>
 	/// <param name="args">The command-line arguments from <c>Main</c>.</param>
 	/// <param name="initialize">Called to initialize the build.</param>
+	/// <param name="settings">The build runner settings.</param>
 	/// <returns>The exit code for the build.</returns>
-	public static int Execute(string[] args, Action<BuildApp> initialize) =>
-		ExecuteAsync(args, initialize).GetAwaiter().GetResult();
+	public static int Execute(string[] args, Action<BuildApp> initialize, BuildRunnerSettings? settings = null) =>
+		ExecuteAsync(args, initialize, settings).GetAwaiter().GetResult();
 
 	/// <summary>
 	/// Executes an automated build. Called from <c>Main</c>.
 	/// </summary>
 	/// <param name="args">The command-line arguments from <c>Main</c>.</param>
 	/// <param name="initialize">Called to initialize the build.</param>
+	/// <param name="settings">The build runner settings.</param>
 	/// <returns>The exit code for the build.</returns>
-	public static async Task<int> ExecuteAsync(string[] args, Action<BuildApp> initialize)
+	public static async Task<int> ExecuteAsync(string[] args, Action<BuildApp> initialize, BuildRunnerSettings? settings = null)
 	{
+		settings ??= new BuildRunnerSettings();
+
 		ArgumentNullException.ThrowIfNull(args);
 		ArgumentNullException.ThrowIfNull(initialize);
 
-		if (Assembly.GetEntryAssembly()?.EntryPoint?.ReturnType == typeof(void))
+		if (!settings.AllowVoidEntrypoint && Assembly.GetEntryAssembly()?.EntryPoint?.ReturnType == typeof(void))
```

Comment:
Maybe we should just delete this surprising behavior. I forgot I added it. Long term, an analyzer would be a better solution to the problem of forgetting to return the exit code.

Reply:
Deleting the behavior would certainly remove any API changes and allow more flexibility in other repositories. If we're comfortable with that, that would likely be my preferred direction.

Reply:
It's fine by me.

Reply:
In that case, I'm going to drop the API changes from this proposal.